src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -405,9 +405,9 @@ private bool IsTagRecoveryStopPoint(HtmlSymbol sym)
 
         private void TagContent()
         {
-            if (!At(HtmlSymbolType.WhiteSpace))
+            if (!At(HtmlSymbolType.WhiteSpace) && !At(HtmlSymbolType.NewLine))
```

Comment:
Suggest checking around for uses of `HtmlSymbolType.WhiteSpace` alone that seem suspect. Then test a bit.